### PR TITLE
Provider JWT Service

### PR DESCRIPTION
**Pull request by arijitAD**
---
name: Pull Request
about: Submit a pull request
title: ''
labels: ''
assignees: '@boz @gosuri'

---

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/ovrclk/akash/blob/master/CONTRIBUTING.md))
- [ ] My issue has received approval from the maintainers or lead with the `design/approved` label

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I've read the [CONTRIBUTION](https://github.com/ovrclk/akash/blob/master/CONTRIBUTING.md) guide
- [ ] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by hydrogen18 — APPROVED** [review]

**Comment by hydrogen18** [comment]
This looks good to me now. Let's get this merged to master so we can develop against it and test it out. If we need to make more adjustments it should be easy now since the JWT server is it's own service.

File: _run/common-commands.mk [hunk 1]
```diff
@@ -55,6 +59,18 @@ provider-update:
 provider-status:
 	$(AKASH) provider status $(PROVIDER_ADDRESS)
 
+.PHONY: jwt-server-authenticate
+jwt-server-authenticate:
+	$(AKASH) provider jwt-server-authenticate \
```
Comment (boz) [inline _run/common-commands.mk:64]: let's just call this command `authenticate` - `akash provider authenticate`.

File: _run/common-commands.mk [hunk 2]
```diff
@@ -55,6 +59,18 @@ provider-update:
...
+		--from      "$(KEY_ADDRESS)" \
+		--provider  "$(PROVIDER_ADDRESS)"
+
+.PHONY: run-jwt-server
+run-jwt-server:
+	$(AKASH) provider run-jwt-server \
```
Comment (boz) [inline _run/common-commands.mk:70]: let's call this command `auth-server` - `akash provider auth-server`.

File: provider/cmd/jwt_server_run.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	"github.com/ovrclk/akash/cmd/common"
+	gwrest "github.com/ovrclk/akash/provider/gateway/rest"
+	cmodule "github.com/ovrclk/akash/x/cert"
+	cutils "github.com/ovrclk/akash/x/cert/utils"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+const (
+	FlagJWTGatewayListenAddress = "jwt-gateway-listen-address"
```
Comment (boz) [inline provider/cmd/jwt_server_run.go:28]: `jwt-auth-listen-address`

**Review by boz — CHANGES_REQUESTED** [review]

File: _run/common-commands.mk (hunk 1 above)
Comment (arijitAD) [reply _run/common-commands.mk]: Done

File: _run/common-commands.mk (hunk 2 above)
Comment (arijitAD) [reply _run/common-commands.mk]: Done

File: provider/cmd/jwt_server_run.go (hunk 3 above)
Comment (arijitAD) [reply provider/cmd/jwt_server_run.go]: Done

**Pull request merged**